src/python/pants/backend/jvm/targets/jvm_target.py

```diff
@@ -90,6 +100,10 @@ def __init__(self,
       'strict_deps': PrimitiveField(strict_deps),
       'fatal_warnings': PrimitiveField(fatal_warnings),
       'zinc_file_manager': PrimitiveField(zinc_file_manager),
+      'javac_plugins': SetOfPrimitivesField(javac_plugins),
```

Comment:
Well, order cannot matter in javac (the plugin options go into an unordered map, so the order is lost almost immediately). They could hypothetically matter in scalac (the plugins are loaded in cmd-line order, as far as I can tell), but they don't in practice - the loading order doesn't affect when each plugin runs - that happens at whatever phase the plugin asked to be run in.

So now it's down to whether we want to err on the side of hypothetical caution, at the expense of triggering spurious invalidation. I just try to avoid spurious invalidation whenever possible. 

We can always change this in the future, should we discover that order matters. That would be a backwards-compatible change, while the reverse would not.

Thoughts?